### PR TITLE
implemented text parser plugin for MacOS launchd.log

### LINKED ISSUE #4685: Text parser plugin for MacOS launchd.log
**Describe the problem:**

I want to create a text parser plugin for the launchd.log file obtained from MacOS sysdiagnose dumps. It can be found at logs/launchd. The file can overflow into launchd.log.# . Where # is a number.

### PR DESCRIPTION
**Pull request by rick-slin**
## One line description of pull request
implemented text parser plugin for MacOS launchd.log

## Description:
implemented text parser plugin for MacOS launchd.log

**Related issue (if applicable):** fixes #4685

## Notes:
All contributions to Plaso undergo [code review](https://github.com/log2timeline/l2tdocs/blob/main/process/Code%20review%20process.asciidoc).
This makes sure that the code has appropriate test coverage and conforms to the
[Plaso style guide](https://plaso.readthedocs.io/en/latest/sources/developer/Style-guide.html).

One of the maintainers will examine your code, and may request changes. Check off the items below in
order, and then a maintainer will review your code.

## Checklist:
* [x] Automated checks (GitHub Actions, AppVeyor) pass
* [x] No new [new dependencies](https://plaso.readthedocs.io/en/latest/sources/developer/Adding-a-new-dependency.html) are required or l2tdevtools has been updated
* [x] Reviewer assigned

### CODE REVIEW

**Comment by rick-slin** [comment]
Could someone look at this PR again?

**Comment by joachimmetz** [comment]
@rick-slin thanks for the reminder I'll have a look when time permits

**Comment by joachimmetz** [comment]
Rebased with HEAD of origin main and fixed merge conflict

File: data/formatters/macos.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ short_source: 'PLIST'
 source: 'MacOS Airport plist'
 ---
 type: 'conditional'
-data_type: 'macos:appfirewall_log:entry'
+data_type: 'macos:appfirewall_log:e:ntry'
```
Comment (joachimmetz) [inline data/formatters/macos.yaml:76]: spurious `:`

**Comment by rick-slin** [comment]
I don't understand why pylint gives the "too many arguments" error. It appears to be from the lambda function in setParseAction.

**Comment by joachimmetz** [comment]
Try changing to the pyparsing 3 preferred API calls e.g. `set_parse_action` https://pyparsing-docs.readthedocs.io/en/latest/pyparsing.html#pyparsing.ParserElement.set_parse_action

I think the linter warning is a side effect how backwards compatibility is done by parparsing.

File: plaso/parsers/text_plugins/macos_launchd.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    event_data = MacOSLaunchdEventData()
+    event_data.body = self._GetValueFromStructure(structure, 'body')
+    event_data.process_name = self._GetValueFromStructure(
+        structure, 'process_name')
+    event_data.severity = self._GetValueFromStructure(structure, 'severity')
+    event_data.written_time = (
+        dfdatetime_time_elements.TimeElementsInMicroseconds(
+            time_elements_tuple=time_elements_structure))
+
+    self._event_data = event_data
```
Comment (joachimmetz) [inline plaso/parsers/text_plugins/macos_launchd.py:119]: @rick-slin why keep `event_data` around and not "produce" it `parser_mediator.ProduceEventData()` ? Are you planning to make changes to extend event_data e.g. for multi line log format /

File: tests/parsers/text_plugins/macos_launchd.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    plugin = macos_launchd.MacOSLaunchdLogTextPlugin()
+    storage_writer = self._ParseTextFileWithPlugin(
+        ['macos_launchd.log'], plugin)
+
+    number_of_event_data = storage_writer.GetNumberOfAttributeContainers(
+      'event_data')
+    self.assertEqual(number_of_event_data, 36609)
+
+    number_of_warnings = storage_writer.GetNumberOfAttributeContainers(
+      'extraction_warning')
```
Comment (joachimmetz) [inline tests/parsers/text_plugins/macos_launchd.py:26]: style nit: use 4 space continuation indentation

**Comment by joachimmetz** [comment]
```
PYTHONPATH=. python3 tests/parsers/text_plugins/macos_launchd.py 
.
----------------------------------------------------------------------
Ran 1 test in 14.330s

OK
```

**Comment by joachimmetz** [comment]
after some tweaks

```
PYTHONPATH=. python3 tests/parsers/text_plugins/macos_launchd.py 
.
----------------------------------------------------------------------
Ran 1 test in 7.382s

OK
```

**Review by joachimmetz — APPROVED** [review]
LGTM

**Pull request merged**